__tests__/proxy.test.ts

```diff
@@ -39,72 +39,112 @@ describe('proxy', () => {
         })
     })
 
-    it('does not return proxyUrl if variables not set', () => {
+    it('getProxyUrl does not return proxyUrl if variables not set', () => {
         let proxyUrl = pm.getProxyUrl(url.parse('https://github.com'));
         expect(proxyUrl).toBeUndefined();
     })
 
-    it('returns proxyUrl if https_proxy set for https url', () => {
+    it('getProxyUrl returns proxyUrl if https_proxy set for https url', () => {
         process.env["https_proxy"] = "https://myproxysvr";
         let proxyUrl = pm.getProxyUrl(url.parse('https://github.com'));
         expect(proxyUrl).toBeDefined();
     })
 
-    it('does not return proxyUrl if http_proxy set for https url', () => {
+    it('getProxyUrl does not return proxyUrl if http_proxy set for https url', () => {
         process.env["http_proxy"] = "https://myproxysvr";
         let proxyUrl = pm.getProxyUrl(url.parse('https://github.com'));
         expect(proxyUrl).toBeUndefined();
     })
     
-    it('returns proxyUrl if http_proxy set for http url', () => {
+    it('getProxyUrl returns proxyUrl if http_proxy set for http url', () => {
         process.env["http_proxy"] = "http://myproxysvr";
         let proxyUrl = pm.getProxyUrl(url.parse('http://github.com'));
         expect(proxyUrl).toBeDefined();
     })
 
-    it('does not return proxyUrl if only host as no_proxy list', () => {
+    it('getProxyUrl does not return proxyUrl if https_proxy set and in no_proxy list', () => {
```

Comment:
Many of the existing no_proxy tests moved down to the section that exercises the `checkBypass` function.

I kept a few basic tests here:
1. https_proxy in no_proxy
2. https_proxy not in no_proxy
3. http_proxy in no_proxy
4. http_proxy not in no_proxy